tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestDataTest.java

```diff
@@ -0,0 +1,236 @@
+package org.unicode.cldr.unittest;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.regex.Pattern;
+import org.unicode.cldr.util.CLDRConfig;
+import org.unicode.cldr.util.CLDRPaths;
+import org.unicode.cldr.util.Factory;
+import org.unicode.cldr.util.SupplementalDataInfo;
+import org.unicode.cldr.util.personname.PersonNameFormatter;
+import org.unicode.cldr.util.personname.PersonNameFormatter.FormatParameters;
+import org.unicode.cldr.util.personname.SimpleNameObject;
+
+public class TestDataTest extends TestFmwkPlus {
+
+    static final boolean SHOW_PROGRESS = System.getProperty("TestDataTest:SHOW_PROGRESS") != null;
+    static final String FILE_FILTER = System.getProperty("TestDataTest:FILE_FILTER");
+
+    private static final String TEST_DATA_DIR = CLDRPaths.COMMON_DIRECTORY + "testData/";
+
+    static CLDRConfig testInfo = CLDRConfig.getInstance();
+    static Factory FACTORY = testInfo.getCldrFactory();
+    private static final SupplementalDataInfo SUPPLEMENTAL_DATA_INFO =
+            testInfo.getSupplementalDataInfo();
+    static final Splitter SEMI_SPLIT = Splitter.on(';').trimResults();
+    static final Splitter COMMA_SPLIT = Splitter.on(',').trimResults();
+
+    public static void main(String[] args) {
+        new TestDataTest().run(args);
+    }
+
+    /** List of directories in testData/ that are tested. Each test adds the ones it handles. */
+    static final Set<Path> FILES_CHECKED = new TreeSet<>();
+
+    /** See GeneratePersonNameTestData */
+    public void testPersonNameTests() {
+        if (SHOW_PROGRESS) {
+            System.out.println();
+        }
+        Path PERSON_NAMES_DIR = Paths.get(TEST_DATA_DIR + "personNameTest");
+        try (DirectoryStream<Path> filepath = Files.newDirectoryStream(PERSON_NAMES_DIR)) {
+            filepath.forEach(x -> checkPersonNameTests(x));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        FILES_CHECKED.add(PERSON_NAMES_DIR);
+    }
+
+    private void checkPersonNameTests(Path filePath) {
+        String name = filePath.toFile().getName();
+        if (name.startsWith("_") || !name.endsWith(".txt")) {
```

Comment:
What are we using `_` at the beginning of the filename for?

Reply:
Those are for files like the readme.

Reply:
I see.  Thanks for the explanation.